### PR TITLE
AM-77-add-argon2id-support-to-am-

### PR DESCRIPTION
**Pull request by farmborough**
## :id: Reference related issue. 
Fixes [AM-77](https://graviteecommunity.atlassian.net/browse/AM-77)

## :pencil2: 

Adds Spring's Argon2 encrypter to AM and `PasswordHistoryService`.

### CODE REVIEW

File: gravitee-am-service/src/main/java/io/gravitee/am/service/authentication/crypto/password/Argon2IdPasswordEncoder.java
Comment (leleueri) [inline gravitee-am-service/src/main/java/io/gravitee/am/service/authentication/crypto/password/Argon2IdPasswordEncoder.java:25]: IHMO, we should make the configuration the values for the Argon2PasswordEncoder to give some flexibilitty to the admin and set by default the [OWASP recommendation](https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html).   

> Use [Argon2id](https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html#argon2id) with a minimum configuration of 15 MiB of memory, an iteration count of 2, and 1 degree of parallelism.

**Review by leleueri — CHANGES_REQUESTED** [review]

**Comment by farmborough** [comment]
Will need to make sure the SCIM config also uses the Argon2 `PasswordEncoder`.

**Review by leleueri — COMMENTED** [review]
Looks good, for the encoder part.
Will approved this PR once the business logic one will be approved.

**Review by leleueri — APPROVED** [review]

**Pull request merged**
